inference-engine/samples/benchmark_app/main.cpp

```diff
@@ -324,8 +330,37 @@ int main(int argc, char *argv[]) {
 
             slog::info << "Loading network files" << slog::endl;
 
+            CNNNetwork cnnNetwork;
+
             auto startTime = Time::now();
-            CNNNetwork cnnNetwork = ie.ReadNetwork(FLAGS_m);
+
+            if (FLAGS_mmap) {
+                // Read XML file content into a string
+                std::ifstream xml_file(FLAGS_m);
+                std::string xml_content((std::istreambuf_iterator<char>(xml_file)), std::istreambuf_iterator<char>());
+#if defined(ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+                std::wstring bin_file = std::wstring_convert<std::codecvt_utf8<wchar_t>>().from_bytes(FLAGS_m);
+                bin_file = bin_file.substr(0, FLAGS_m.find_last_of('.')) + L".bin";
+#else
+                std::string bin_file = FLAGS_m.substr(0, FLAGS_m.find_last_of('.')) + ".bin";
+#endif
+
+                // Get file size, size is needed to create a tensor descriptor
+                std::ifstream bin_stream(bin_file, std::ios::in);
+                bin_stream.seekg(0, std::ios::end);
+                size_t bin_size = bin_stream.tellg();
+                bin_stream.close();
```

Comment:
Let's create a simple function like `fileSize` in `samples/common.h` to simplify the sample's code.